server/src/gameserver/sdk.rs

```diff
@@ -12,32 +12,17 @@ pub struct Manager {
     _health_checker: HealthChecker,
 }
 
-type BoardByUsers = HashMap<usize, HashSet<usize>>;
+type UsersByBoard = HashMap<usize, HashSet<usize>>;
 
-impl Manager {
-    pub async fn setup() -> Result<Manager, Error> {
-        log::info!("connecting to agones sdk sidecar...");
-        let sdk = Sdk::new(None, None)
-            .await
-            .map_err(|e| Error::SetupFailure(format!("{}", e)))?;
-
-        let health_checker = HealthChecker::new(&sdk);
-        let board_events = Manager::new_spaces_channel(&sdk);
-        Ok(Manager {
-            _health_checker: health_checker,
-            board_events,
-        })
-    }
-
-    fn new_spaces_channel(sdk: &Sdk) -> mpsc::Sender<BoardEvent> {
-        let (tx, rx) = mpsc::channel::<BoardEvent>(10);
-
-        task::spawn(Self::board_event_handler(sdk.clone(), rx));
-
-        tx
-    }
+struct BoardEventHandler {
+    sdk: Sdk,
+    boards_map: UsersByBoard,
+    max_board_count: usize,
+    max_user_count: usize,
+}
 
-    async fn board_event_handler(mut sdk: Sdk, mut rx: mpsc::Receiver<BoardEvent>) {
+impl BoardEventHandler {
+    pub fn from_env(sdk: Sdk) -> BoardEventHandler {
         let max_board_count = std::env::var("BOARD_CAPACITY")
             .map(|c| c.parse::<usize>().expect("Invalid board capacity"))
             .unwrap_or(1);
```

Comment:
Should be 2 now? To keep up with `ENV BOARD_CAPACITY "2"` setting?

Reply:
(with users set to 200). Just wondering if the defaults need to match.

Reply:
they don't need to match - the fallback values exist to avoid panic and help with local dev (when not running inside k8s, maybe with an external agones sdk instance)